### PR TITLE
Modify error handling to handle errors in one place

### PR DESCRIPTION
**Pull request by liorLahav**

### CODE REVIEW

File: src/TelegramBot/Bot.ts
Comment (liorLahav) [inline src/TelegramBot/Bot.ts:1]: the greetings call don't get the error handling treatment

Comment (liorLahav) [reply src/TelegramBot/Bot.ts:1]: I think there is unnecessary code duplication here.

File: src/models/MessageHandler.ts [hunk 1]
```diff
@@ -24,50 +24,66 @@ export class MessageHandler {
...
+        userProfile.personalDetails = this.parseMarkdownToJson(res);
+        this.userStore.saveUser(userProfile);
+      } catch (error) {
+        logger.error("Can't parse message:", error);
+        throw error; // Re-throw the error to be handled by the calling function
+      }
     } catch (error) {
-      console.log("can't parse message");
+      logger.error("Can't parse message:", error);
+      throw error; // Re-throw the error to be handled by the calling function
```
Comment (liorLahav) [inline src/models/MessageHandler.ts:86]: You have an inner try catch in the main try catch that handles errors the same way

File: src/models/MessageHandler.ts
Comment (liorLahav) [inline src/models/MessageHandler.ts:1]: the greetings call don't get the error handling treatment

**Comment by TheZohan** [comment]
@liorLahav I fixed the retry code to remove duplication. Can you please review?

File: src/TelegramBot/Bot.ts [hunk 2]
```diff
@@ -51,23 +57,33 @@ export const initializeBot = async (): Promise<Telegraf> => {
...
-      const message = JSON.stringify(
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        (error as any)?.response?.data?.error ?? 'Unable to extract error',
-      );
+    } while (retries < maxRetries && !success);
 
-      await ctx.reply('Whoops! There was an error while talking to OpenAI. Error: ' + message);
+    if (!success) {
+      // All retries failed, inform the user
+      const errorMessage = i18n.t('errorMessage');
```
Comment (liorLahav) [inline src/TelegramBot/Bot.ts:82]: What does t do? i think the name is unclear

**Pull request merged**